zio-json/shared/src/main/scala/zio/json/JsonCodecConfiguration.scala

```diff
@@ -14,33 +20,77 @@ import zio.json.JsonCodecConfiguration.SumTypeHandling.WrapperWithClassNameField
  *   see [[jsonNoExtraFields]]
  * @param sumTypeMapping
  *   see [[jsonHintNames]]
- * @param explicitNulls
- *   see [[jsonExplicitNull]]
  */
 final case class JsonCodecConfiguration(
   sumTypeHandling: SumTypeHandling = WrapperWithClassNameField,
   fieldNameMapping: JsonMemberFormat = IdentityFormat,
   allowExtraFields: Boolean = true,
   sumTypeMapping: JsonMemberFormat = IdentityFormat,
-  explicitNulls: Boolean = false
+  explicitNulls: Boolean = false,
+  explicitEmptyCollections: ExplicitEmptyCollections = ExplicitEmptyCollections()
 ) {
+  def this(
+    sumTypeHandling: SumTypeHandling,
+    fieldNameMapping: JsonMemberFormat,
+    allowExtraFields: Boolean,
+    sumTypeMapping: JsonMemberFormat,
+    explicitNulls: Boolean
+  ) = this(
+    sumTypeHandling,
+    fieldNameMapping,
+    allowExtraFields,
+    sumTypeMapping,
+    explicitNulls,
+    ExplicitEmptyCollections()
+  )
 
   def copy(
     sumTypeHandling: SumTypeHandling = WrapperWithClassNameField.asInstanceOf[SumTypeHandling],
     fieldNameMapping: JsonMemberFormat = IdentityFormat.asInstanceOf[JsonMemberFormat],
     allowExtraFields: Boolean = true,
     sumTypeMapping: JsonMemberFormat = IdentityFormat.asInstanceOf[JsonMemberFormat],
-    explicitNulls: Boolean = false
+    explicitNulls: Boolean = false,
+    explicitEmptyCollections: ExplicitEmptyCollections = ExplicitEmptyCollections()
+  ) = new JsonCodecConfiguration(
+    sumTypeHandling,
+    fieldNameMapping,
+    allowExtraFields,
+    sumTypeMapping,
+    explicitNulls,
+    explicitEmptyCollections
+  )
+
+  def copy(
+    sumTypeHandling: SumTypeHandling,
+    fieldNameMapping: JsonMemberFormat,
+    allowExtraFields: Boolean,
+    sumTypeMapping: JsonMemberFormat,
+    explicitNulls: Boolean
   ) = new JsonCodecConfiguration(
     sumTypeHandling,
     fieldNameMapping,
     allowExtraFields,
     sumTypeMapping,
-    explicitNulls
+    explicitNulls,
+    ExplicitEmptyCollections()
```

Comment:
Shouldn't it be `this.explicitEmptyCollections` instead of resetting to default `ExplicitEmptyCollections()`?

Reply:
yes